src/functions-and-arrays.js

```diff
@@ -1,38 +1,91 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
-
+function maxOfTwoNumbers(n1, n2) {
+  if (n1 > n2) {
+      return n1;
+    } else {
+      return n2;
+  }
+}
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
+function findLongestWord(words) {
+  if (words.length === 0) {
+    return null;
+  }
+  let longest = words[0];
+  for (let i = 0; i < words.length; i++) {
+    if (words[i].length > longest.length) {
+      longest = words[i]; 
+    }
+  }
+  return longest;
+}
 
 
 
 // Iteration #3: Calculate the sum
-const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
-
-function sumNumbers() {}
 
+function sumNumbers(numbers) {
+  
+  let sum = 0;
+  for (let i = 0; i < numbers.length; i++) {
+    const currentNumber = numbers[i]
+    console.log(currentNumber);
+    sum += currentNumber
+  } 
+  return sum
+}
 
 
 // Iteration #3.1 Bonus:
-function sum() {}
+const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sum(numbers) {
+  let sum = 0;
+  for (let i = 0; i < numbers.length; i++) {
+  currentNumber = numbers[i]
+  sum += currentNumber
+ } 
+ return sum
+}
 
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function averageNumbers(numbersAvg) {
+let average = 0;
+if (numbersAvg = [""]) {
+  return null
+}
+ for (let i = 0; i < numbersAvg.length; i++) {
+  const element = numbersAvg[i];
+  average += element
+  return average / numbersAvg.length 
+ }
+}
 
 
 // Level 2: Array of strings
 const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
 
-function averageWordLength() { }
+function averageWordLength(wordsArray) {
+  let wordAverage = 0;
+
+  if (wordsArray.length === 0) {
+    return null;
+  }
+ for (let i = 0; i < wordsArray.length; i++) {
+  const word = wordsArray[i].length;
+  wordAverage += word
+  return wordAverage / wordsArray.length
+ } 
```

Comment:
```suggestion
 for (let i = 0; i < wordsArray.length; i++) {
  const word = wordsArray[i].length;
  wordAverage += word
 } 
 return wordAverage / wordsArray.length
```
Lo mismo que antes :D

Reply:
Gracias por los comentarios Cristian, voy a seguir practicando